dqc_us_rules/dqc_us_0001.py

```diff
@@ -26,6 +28,128 @@
 _UGT_FACT_KEY = 'ugt_fact'
 _NO_FACT_KEY = 'no_fact'
 _EXT_FACT_KEY = 'ext_fact'
+_EARLIEST_US_GAAP_YEAR = 2014
+_CONFIG_JSON_FILE = os.path.join(
+            os.path.dirname(__file__),
+            'resources',
+            'DQC_US_0001',
+            'dqc_0001.json'
+        )
+
+ugtDocs = (
+    {
+        "year": 2014,
+        "namespace": "http://fasb.org/us-gaap/2014-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/elts/us-gaap-doc-2014-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/entire/us-gaap-entryPoint-std-2014-01-31.xsd",  # noqa
+    },
+    {
+        "year": 2015,
+        "namespace": "http://fasb.org/us-gaap/2015-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/elts/us-gaap-doc-2015-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/entire/us-gaap-entryPoint-std-2015-01-31.xsd",  # noqa
+    },
+    {
+        "year": 2016,
+        "namespace": "http://fasb.org/us-gaap/2016-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/elts/us-gaap-doc-2016-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/entire/us-gaap-entryPoint-std-2016-01-31.xsd",  # noqa
+    },
+)
+
+
+def _tr_mem(val,
+            ugt,
+            parent_model_object,
+            rel_name,
+            elr):
+    """
+    Walks the taxonomy for a given axis
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :param ugt: dict of taxonomies and their entirance points
+    :type ugt: dict
+    :param parent_model_object: model object for Axis
+    :type parent_model_object:class:'~arelle.ModelDTSObject.ModelConcept'
+    :param rel_name: The role for the relationship
+    :type rel_name: str
+    :param elr: Linkrole
+    :type elr: str
+    :return: list of members for the axis specified
+    :rtype: dict
+    """
+    axMem = set()
+    cntlr = val.modelXbrl.modelManager.cntlr
+    ugt_entry_xsd = ugt["entryXsd"]
+    dm_ld_inst = ModelXbrl.load(
+        val.modelXbrl.modelManager, openFileSource(ugt_entry_xsd, cntlr),
+        ("built us-gaap member cache"))
+    rels = dm_ld_inst.relationshipSet(rel_name,
```

Comment:
SG this is a little bit of a funky line but I think this would fit our guide better:
```
    rels = dm_ld_inst.relationshipSet(
        rel_name, elr
    ).fromModelObject(parent_model_object)
```